File: examples/H20/robots/sdk_package/example_py/h2x/high_level/walk_run_mode.py
import humanoid_sdk_py.h2x as h2x
import time

h2x_high_level = h2x.H2xHighLevel()
time.sleep(1)

print("=== h2x 进入走跑模式示例 ===")
ret = h2x_high_level.WalkRunMode()
if ret == 0:
    print("进入走跑模式请求成功")
else:
    print("进入走跑模式请求失败，错误码：",ret)

ret = h2x_high_level.waitRunning(5000)
if ret == 0:
    print("模式切换成功")
else:
    print("模式切换失败，错误码：",ret)

File: examples/H20/robots/sdk_package/example/h2x/high_level/walk_run_mode.cpp
#include <iostream>

#include "h2x/h2x_high_level.h"

using namespace humanoid;
using namespace humanoid::h2x;

int main() {
  std::cout << "=== h2x 进入走跑模式示例 ===" << std::endl;
  H2xHighLevel high_level;

  int ret = high_level.WalkRunMode();
  if (ret == 0) {
    std::cout << "进入走跑模式请求成功" << std::endl;
  } else {
    std::cout << "进入走跑模式请求失败, 错误码：" << ret << std::endl;
    return -1;
  }

  ret = high_level.waitRunning(5000);
  if (ret == 0) {
    std::cout << "模式切换成功" << std::endl;
  } else {
    std::cout << "模式切换失败, 错误码：" << ret << std::endl;
    return -1;
  }
  return 0;
}